internal/provider/resource_nd_site.go

```diff
@@ -0,0 +1,489 @@
+	resp.Diagnostics.Append(req.State.Get(ctx, &stateData)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("Update of resource nd_site with id '%s'", data.Id.ValueString()))
+
+	jsonPayload := getSiteCreateJsonPayload(ctx, &resp.Diagnostics, data)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	DoRestRequest(ctx, &resp.Diagnostics, r.client, fmt.Sprintf("%s/%s", sitePath, data.Id.ValueString()), "PUT", jsonPayload)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	setSiteId(ctx, data)
+
+	getAndSetSiteAttributes(ctx, &resp.Diagnostics, r.client, data)
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	tflog.Debug(ctx, "End update of resource nd_site")
+}
+
+func (r *SiteResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	tflog.Debug(ctx, "Start delete of resource: nd_site")
+	var data *SiteResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("Delete of resource nd_site with id '%s'", data.Id.ValueString()))
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	DoRestRequest(ctx, &resp.Diagnostics, r.client, fmt.Sprintf("%s/%s", sitePath, data.Id.ValueString()), "DELETE", nil)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	tflog.Debug(ctx, fmt.Sprintf("End delete of resource nd_site with id '%s'", data.Id.ValueString()))
+}
+
+func (r *SiteResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	tflog.Debug(ctx, "Start import state of resource: nd_site")
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+
+	var stateData *SiteResourceModel
+	resp.Diagnostics.Append(resp.State.Get(ctx, &stateData)...)
+
+	// The API does not return the username, password, and login_domain attributes.
+	// Therefore, these attributes will be assigned based on the values of environment variables.
+	username := os.Getenv("ND_SITE_USERNAME")
+	if username == "" {
+		resp.Diagnostics.AddError("Missing input", "The username of the ND site must be provided during import, please set the ND_SITE_USERNAME environment variable")
+	}
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("username"), username)...)
+
+	password := os.Getenv("ND_SITE_PASSWORD")
+	if password == "" {
+		resp.Diagnostics.AddError("Missing input", "The password of the ND site must be provided during import, please set the ND_SITE_PASSWORD environment variable")
+	}
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("password"), password)...)
+
+	loginDomain := os.Getenv("ND_SITE_LOGIN_DOMAIN")
+	if loginDomain == "" {
+		resp.Diagnostics.AddError("Missing input", "The login_domain of the ND site must be provided during import, please set the ND_SITE_LOGIN_DOMAIN environment variable")
+	}
```

Comment:
Do we need this? You ignore it anyway after the create so why do we actually need it? Can't we use a ModifyPlan to take care of it at the next plan?